cmd/sync-tags/main.go

```diff
@@ -323,6 +306,31 @@ func main() {
 			glog.Fatalf("Failed to create tag %q: %v", bName, err)
 		}
 		createdTags = append(createdTags, bName)
+
+		// if publishSemverTags is true, create annotated semver tag
+		if *publishSemverTags {
+			semverTag := name[:1] + "0" + name[2:] // replace v1.x.y with v0.x.y
+
+			if goModChanged && len(dependentRepos) > 0 {
+				wt := checkoutBranchTagCommit(r, bh, dependentRepos)
+				changed = updateGoModAndGodeps(semverTag, dependentRepos, *generateGodeps, true)
```

Comment:
by creating the prefixed tag matching upstream (kubernetes-1.x.y) first, and the semver tag second, isn't this no longer re-entrant if the script fails during the second tag creation (since we return early if bTagCommits[bName] already exists)?

Reply:
also, won't calling updateGoModAndGodeps again mean that we'll change the go.mod files before adding this tag, so `kubernetes-1.x.y` and `v0.x.y` tags will not actually point to the same sha? I expected both tags to point to the same content

Reply:
>  isn't this no longer re-entrant if the script fails during the second tag creation (since we return early if bTagCommits[bName] already exists)?

will fix

> also, won't calling updateGoModAndGodeps again mean that we'll change the go.mod files before adding this tag, so kubernetes-1.x.y and v0.x.y tags will not actually point to the same sha? I expected both tags to point to the same content

My understanding is that we'd want to use `v0.x.y` tags in go.mod while publishing the `v0.x.y` tags (example - https://github.com/prowtest/client-go/blob/v0.17.0-rc.2/go.mod#L41-L42).

To keep the sha same for both `kubernetes-1.x.y` and `v0.x.y` (makes sense), should we then also use `v0.x.y` tags in go.mod while publishing `kubernetes-1.x.y` i.e. same go.mod content for `kubernetes-1.17.0-rc.2` as the link above?

Reply:
> To keep the sha same for both kubernetes-1.x.y and v0.x.y (makes sense), should we then also use v0.x.y tags in go.mod while publishing kubernetes-1.x.y i.e. same go.mod content for kubernetes-1.17.0-rc.2 as the link above and we won't use pseudoversions in tags anymore?

That was my expectation

Reply:
Updated.